### PR TITLE
docs: edits to the dm join article

### PR DESCRIPTION
**Pull request by IndrajeetPatil**

### CODE REVIEW

File: vignettes/tech-dm-join.Rmd [hunk 1]
```diff
@@ -26,7 +26,7 @@ Read more about relational data models in the vignette ["Introduction to Relatio
 
 First, we load the packages that we need:
 
-```{r message=FALSE, warning=FALSE}
```
Comment (IndrajeetPatil) [inline vignettes/tech-dm-join.Rmd:29]: To be consistent with all the other vignettes, this is taken care of in `setup.R`.

File: vignettes/tech-dm-join.Rmd [hunk 2]
```diff
@@ -128,15 +128,10 @@ An example with filtering on a `dm` and then merging:
...
 
-Currently, it is important to call `dm_apply_filters()` after piping your conditions.
-Only then the underlying tables and key relations are updated and you can perform a join on the filtered data.
-We are working towards removing this inconvenience [#62](https://github.com/cynkra/dm/issues/62).
-
 ```{r}
 dm_nycflights13() %>%
   dm_filter(airlines, name == "Delta Air Lines Inc.") %>%
   dm_filter(flights, month == 5) %>%
-  dm_apply_filters() %>%
```
Comment (IndrajeetPatil) [inline vignettes/tech-dm-join.Rmd:139]: AFAIK, no longer necessary.

Comment (krlmlr) [reply vignettes/tech-dm-join.Rmd:139]: This is about to be deprecated.

File: vignettes/tech-dm-join.Rmd [hunk 3]
```diff
@@ -157,4 +152,4 @@ dm_nycflights13() %>%
 ```
 
 Be aware that all column names need to be unique.
-The `dm_flatten_to_tbl` cares about automatically renaming the relevant columns and prints if something was changed, e.g. `name -> airlines.name`.
+The `dm_flatten_to_tbl` takes cares of this by automatically renaming the relevant columns and informs the user if any names were changed, e.g. `name -> airlines.name`.
```
Comment (krlmlr) [inline vignettes/tech-dm-join.Rmd:155]:
```suggestion
The `dm_flatten_to_tbl` takes care of this by automatically renaming the relevant columns and informs the user if any names were changed, e.g. `name -> airlines.name`.
```

Also, the output of the renaming instructions have changed IIRC.

**Review by krlmlr — COMMENTED** [review]
Thanks!

File: vignettes/tech-dm-join.Rmd (hunk 2 above)
Comment (krlmlr) [reply vignettes/tech-dm-join.Rmd:139]: I meant that `dm_join_to_tbl()` will go away. Need to think about how to handle.

File: vignettes/tech-dm-join.Rmd [hunk 4]
```diff
@@ -157,4 +152,4 @@ dm_nycflights13() %>%
 ```
 
 Be aware that all column names need to be unique.
-The `dm_flatten_to_tbl` cares about automatically renaming the relevant columns and prints if something was changed, e.g. `name -> airlines.name`.
+The `dm_flatten_to_tbl` takes care of this by automatically renaming the relevant columns and informs the user if any names were changed, e.g. `name -> airlines.name`.
```
Comment (krlmlr) [inline vignettes/tech-dm-join.Rmd:155]: New way of printing renames?

Comment (maelle) [reply vignettes/tech-dm-join.Rmd:155]: what does this comment mean, was it tackled @IndrajeetPatil?

**Review by maelle — COMMENTED** [review]
only a question :-)

File: vignettes/tech-dm-join.Rmd (hunk 4 above)
Comment (IndrajeetPatil) [reply vignettes/tech-dm-join.Rmd:155]: Yeah, I am not sure about this means either, so, no, it hasn't been tackled 😅

Comment (maelle) [reply vignettes/tech-dm-join.Rmd:155]: ah ok then let's wait for a clarification by @krlmlr :smiley_cat:

Comment (IndrajeetPatil) [reply vignettes/tech-dm-join.Rmd]: @krlmlr I am still awaiting a clarification on this point. Thanks.

**Comment by IndrajeetPatil** [comment]
@maelle I have updated the vignette to mention flatten, squash, and wrap.

File: vignettes/tech-dm-join.Rmd (hunk 4 above)
Comment (krlmlr) [reply vignettes/tech-dm-join.Rmd]: - The `dm_flatten_to_tbl()` function ...
- The format of the renaming messages `name -> airlines.name` has changed

**Comment by IndrajeetPatil** [comment]
I think there are no more outstanding issues/comments here.

File: vignettes/tech-dm-join.Rmd [hunk 5]
```diff
@@ -116,37 +116,32 @@ Another example:
...
 An example with filtering on a `dm` and then merging:
 
 **Get all flights from Delta Air Lines which didn't depart from John F. Kennedy International Airport in May 2013 - and join all the airports data into the `flights` table.**
 
-Currently, it is important to call `dm_apply_filters()` after piping your conditions.
-Only then the underlying tables and key relations are updated and you can perform a join on the filtered data.
-We are working towards removing this inconvenience [#62](https://github.com/cynkra/dm/issues/62).
-
 ```{r}
 dm_nycflights13() %>%
```
Comment (krlmlr) [inline vignettes/tech-dm-join.Rmd:132]: We want the full data here.

```suggestion
dm_nycflights13(subset = FALSE) %>%
```

File: vignettes/tech-dm-join.Rmd [hunk 6]
```diff
@@ -156,5 +151,16 @@ dm_nycflights13() %>%
...
+
+Also, be aware that all column names need to be unique.
+The `dm_flatten_to_tbl()` takes care of this by automatically renaming the relevant columns and informs the user if any names were changed, e.g. `dm_rename(airlines, airlines.name = name)`.
+
+If you want to merge all tables, but get a `dm` object in return, use `dm_wrap_tbl()` instead:
+
+```{r}
+dm_nycflights13() %>%
+  dm_wrap_tbl(root = flights)
+```
```
Comment (krlmlr) [inline vignettes/tech-dm-join.Rmd:165]:
```suggestion
If you want to merge all tables, but get a nested table in return, use `dm_wrap_tbl()` with `pull_tbl()` instead:

```{r}
dm_nycflights13() %>%
  dm_wrap_tbl(root = flights) %>%
  pull_tbl(flights)
```
```

**Review by krlmlr — COMMENTED** [review]
Thanks, good job!

**Comment by IndrajeetPatil** [comment]
Thanks! I have accepted your suggested commits. I don't think there is any outstanding issue here.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**